hpx/lcos/local/recursive_mutex.hpp

```diff
@@ -176,6 +180,8 @@ namespace hpx { namespace lcos { namespace local
             {
                 if (mtx.try_lock())
                 {
+                    util::ignore_lock(&mtx);
+                    //util::register_lock(this);
```

Comment:
When the lock is unlocked again, it's removed from the held locks map and therefor reset.